digdag-ui can't show tasklog when giving hash parameter to for_each>:

When giving hash parameter:
```yaml
+step1:
  echo>: start
+step2:
  +repeat:
    for_each>:
      param:
        - key: p1
        - key: p2
        - key: p3
    _do:
      +substep:
        echo>: ${param.key}
+step3:
  echo>: end
```

Tasks success but digdag-ui fails showing tasklog:

![image](https://user-images.githubusercontent.com/191684/30897780-09de0516-a393-11e7-9561-60f2feddb8fb.png)

Hello, @syucream 

In short, it seems to bug.

In this case, Digdag store task logs into the following files.

```
+test+step2+repeat^sub+for-0=param=0=%7B%22key%22%3A%22+substep@59cb3fd10328b740.11791@hostname.local.log.gz
+test+step2+repeat^sub+for-0=param=1=%7B%22key%22%3A%22+substep@59cb3fd114cd5140.11791@hostname.local.log.gz
+test+step2+repeat^sub+for-0=param=2=%7B%22key%22%3A%22+substep@59cb3fd1270a81c0.11791@hostname.local.log.gz
```

But API try to read the file `+test+step2+repeat^sub+for-0=param=1={"key":"+substep@59cb3fd114cd5140.11791@hostname.local.log.gz`.
The `{"key":}` part was decoded. That's why, exception raised. 


```
2017-09-27 15:06:42 +0900 [ERROR] (XNIO-1 task-7): UT005023: Exception handling request to /api/logs/1/files/+test+step2+repeat^sub+for-0=param=1=%7B%22key%22%3A%22+substep@59cb3fd114cd5140.11791@hostname.local.log.gz
org.jboss.resteasy.spi.UnhandledException: java.lang.RuntimeException: java.nio.file.NoSuchFileException: /tmp/taks-logs/2017-09-27/0.1test@20170927T060607+0000/+test+step2+repeat^sub+for-0=param=1={"key":"+substep@59cb3fd114cd5140.11791@hostname.local.log.gz
	at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:76)
	at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:212)
	at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:165)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:392)
	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:199)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:221)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:56)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:51)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
```

--- a/digdag-ui/model.js
+++ b/digdag-ui/model.js
@@ -299,7 +299,7 @@ export class Model {
 
   fetchLogFile (attemptId: string, file: LogFileHandle) {
     if (!file.direct) {
-      return this.fetchArrayBuffer(`${this.config.url}logs/${attemptId}/files/${file.fileName}`)
+      return this.fetchArrayBuffer(`${this.config.url}logs/${attemptId}/files/${encodeURIComponent(file.fileName)}`)
     }
     return this.fetchArrayBuffer(file.direct)
   }

